tasks/task_abricate_flu.wdl

```diff
@@ -0,0 +1,77 @@
+version 1.0
+
+task abricate_flu {
+  input {
+    File assembly
+    String samplename
+    String database = "insaflu"
+    String nextclade_flu_h1n1_tag
+    String nextclade_flu_h3n2_tag
+    String nextclade_flu_vic_tag
+    String nextclade_flu_yam_tag
+    Int minid = 70
+    Int mincov =60
+    Int cpu = 2
+    Int memory = 4
+    String docker = "staphb/abricate:1.0.1-insaflu-220727"
+  }
+  command <<<
+    date | tee DATE    
+    abricate -v | tee ABRICATE_VERSION
+    # run abricate
+    abricate \
+      --db ~{database} \
+      ~{'--minid ' + minid} \
+      ~{'--mincov ' + mincov} \
+      --threads ~{cpu} \
+      --nopath \
+      ~{assembly} > ~{samplename}_abricate_hits.tsv
+    # capturing flu type (A or B) and subtype (e.g. H1 and N1)
+    grep "M1" ~{samplename}_abricate_hits.tsv | awk -F '\t' '{ print $15 }' > FLU_TYPE
+    HA_hit=$(grep "HA" ~{samplename}_abricate_hits.tsv | awk -F '\t' '{ print $15 }')
+    NA_hit=$(grep 'NA' ~{samplename}_abricate_hits.tsv | awk -F '\t' '{ print $15 }')
+    flu_subtype="${HA_hit}${NA_hit}" && echo "$flu_subtype" >  FLU_SUBTYPE
+    # set nextclade variables based on subptype
+    run_nextclade=true
+    touch NEXTCLADE_REF NEXTCLADE_NAME NEXTCLADE_DS_TAG
```

Comment:
Do we actually want to set these NextClade variables in this task? It doesn't feel like the right spot to me (though it is certainly the most convenient).

I'm just thinking about when its time to update these values that this task will not be the first place I would look -- I'd go to taxon_id's nextclade task. Obviously, you be able to find these values here by tracing back the outputs, but it just feels a little wrong to have it here since it has nothing to do with abricate except that we set the nextclade values based on the `flu_subtype` output.

Personally, I think best practice would be to move it into a different task, but I guess we'll have to determine if the effort is worth it haha

Reply:
Ah, great question. It certainly could be its own task in and of itself, but this felt most appropriate as this is where the type/subtype is getting defined which also dictate whether or not Nextclade is run at all--sort of how we set `merlin_tags` in PHBG through the gambit task.